internal/security/setup/option/pkiOption_test.go

```diff
@@ -44,21 +44,37 @@ func TestNewPkiInitOption_GenerateOnly(t *testing.T) {
 	assert.Equal(false, generateOff.(*PkiInitOption).GenerateOpt)
 }
 
-func getMockArguments() []interface{} {
-	var ifc []interface{}
-	// use reflection to find out how many bool type of options in PkiInitOption struct
-	// and create a slice of mock argument interface instances
-	elm := reflect.ValueOf(&PkiInitOption{}).Elem()
-	for i := 0; i < elm.NumField(); i++ {
-		field := elm.Field(i)
-		switch field.Kind() {
-		case reflect.Bool:
-			ifc = append(ifc, mock.AnythingOfTypeArgument("func(*option.PkiInitOption) (option.exitCode, error)"))
-		}
+func TestNewPkiInitOption_CacheOnly(t *testing.T) {
+	assert := assert.New(t)
+
+	options := PkiInitOption{
+		CacheOpt: true,
 	}
+	// cache option given
+	optionsExecutor, _, _ := NewPkiInitOption(options)
+	assert.NotNil(optionsExecutor)
+	assert.Equal(true, optionsExecutor.(*PkiInitOption).CacheOpt)
+
+	// cache option omitted
+	options.CacheOpt = false
+	optionsExecutor, _, _ = NewPkiInitOption(options)
+	assert.NotNil(optionsExecutor)
+	assert.Equal(false, optionsExecutor.(*PkiInitOption).CacheOpt)
+}
 
-	return ifc
+func TestNewPkiInitOption_CacheAndGenerate(t *testing.T) {
+	assert := assert.New(t)
+
+	options := PkiInitOption{
+		GenerateOpt: true,
+		CacheOpt:    true,
+	}
+	optionsExecutor, status, err := NewPkiInitOption(options)
+	assert.Empty(optionsExecutor)
+	assert.Equal(exitWithError.intValue(), status)
```

Comment:
this is another instance where I don't see the value from the `exitCode` type since you just end up comparing it with an int anyways, but this is fine for now